diff --git a/os/os/hal/ports/STM32/STM32F7xx/hal_lld.h b/os/os/hal/ports/STM32/STM32F7xx/hal_lld.h
index 41da829..ab3d414 100644
--- a/os/os/hal/ports/STM32/STM32F7xx/hal_lld.h
+++ b/os/os/hal/ports/STM32/STM32F7xx/hal_lld.h
@@ -57,16 +57,19 @@
  * @{
  */
 #if defined(STM32F722xx) || defined(__DOXYGEN__)
-#define PLATFORM_NAME           "STM32F745 Very High Performance with DSP and FPU"
+#define PLATFORM_NAME           "STM32F722 Very High Performance with DSP and FPU"
 
 #elif defined(STM32F723xx)
-#define PLATFORM_NAME           "STM32F745 Very High Performance with DSP and FPU"
+#define PLATFORM_NAME           "STM32F723 Very High Performance with DSP and FPU"
+
+#elif defined(STM32F730xx)
+#define PLATFORM_NAME           "STM32F730 Very High Performance with DSP and FPU"
 
 #elif defined(STM32F732xx)
-#define PLATFORM_NAME           "STM32F745 Very High Performance with DSP and FPU"
+#define PLATFORM_NAME           "STM32F732 Very High Performance with DSP and FPU"
 
 #elif defined(STM32F733xx)
-#define PLATFORM_NAME           "STM32F745 Very High Performance with DSP and FPU"
+#define PLATFORM_NAME           "STM32F733 Very High Performance with DSP and FPU"
 
 #elif defined(STM32F745xx)
 #define PLATFORM_NAME           "STM32F745 Very High Performance with DSP and FPU"
@@ -902,6 +905,10 @@
 #error "Using a wrong mcuconf.h file, STM32F7xx_MCUCONF not defined"
 #endif
 
+#if defined(STM32F730xx) && !defined(STM32F730_MCUCONF)
+#error "Using a wrong mcuconf.h file, STM32F730_MCUCONF not defined"
+#endif
+
 #if defined(STM32F722xx) && !defined(STM32F722_MCUCONF)
 #error "Using a wrong mcuconf.h file, STM32F722_MCUCONF not defined"
 #endif
diff --git a/os/os/hal/ports/STM32/STM32F7xx/stm32_registry.h b/os/os/hal/ports/STM32/STM32F7xx/stm32_registry.h
index df2b063..dd93890 100644
--- a/os/os/hal/ports/STM32/STM32F7xx/stm32_registry.h
+++ b/os/os/hal/ports/STM32/STM32F7xx/stm32_registry.h
@@ -75,10 +75,10 @@
 #endif
 
 /*===========================================================================*/
-/* STM32F722xx, STM32F723xx, STM32F732xx, STM32F733xx.                       */
+/* STM32F722xx, STM32F723xx, STM32F732xx, STM32F733xx, STM32F730xx.          */
 /*===========================================================================*/
 #if defined(STM32F722xx) || defined(STM32F723xx) || defined(STM32F732xx) || \
-    defined(STM32F733xx) || defined(__DOXYGEN__)
+    defined(STM32F733xx) || defined(STM32F730xx) || defined(__DOXYGEN__)
 /* ADC attributes.*/
 #define STM32_ADC_HANDLER                   Vector88
 #define STM32_ADC_NUMBER                    18
